cmd/krew/cmd/search.go

```diff
@@ -43,58 +44,91 @@ Examples:
   To fuzzy search plugins with a keyword:
     kubectl krew search KEYWORD`,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		plugins, err := indexscanner.LoadPluginListFromFS(paths.IndexPluginsPath(constants.DefaultIndexName))
-		if err != nil {
-			return errors.Wrap(err, "failed to load the list of plugins from the index")
+		indexes := []indexoperations.Index{
+			{
+				Name: constants.DefaultIndexName,
+				URL:  constants.IndexURI, // unused here but providing for completeness
+			},
 		}
-		names := make([]string, len(plugins))
-		pluginMap := make(map[string]index.Plugin, len(plugins))
-		for i, p := range plugins {
-			names[i] = p.Name
-			pluginMap[p.Name] = p
+		if os.Getenv(constants.EnableMultiIndexSwitch) != "" {
+			out, err := indexoperations.ListIndexes(paths)
+			if err != nil {
+				return errors.Wrapf(err, "failed to list plugin indexes available")
+			}
+			indexes = out
 		}
 
+		klog.V(3).Infof("found %d indexes", len(indexes))
+
+		var plugins []pluginEntry
+		for _, idx := range indexes {
+			ps, err := indexscanner.LoadPluginListFromFS(paths.IndexPluginsPath(idx.Name))
+			if err != nil {
+				return errors.Wrap(err, "failed to load the list of plugins from the index")
+			}
+			for _, p := range ps {
+				plugins = append(plugins, pluginEntry{p, idx.Name})
+			}
+		}
+
+		keys := func(v map[string]pluginEntry) []string {
+			out := make([]string, 0, len(v))
+			for k := range v {
+				out = append(out, k)
+			}
+			return out
+		}
```

Comment:
There's only a single use of this lambda. Have you considered inlining it?

A proper function would also make sense because this does not capture any variables.

Reply:
will do

Reply:
Done.